CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ include(GNUInstallDirs)
 include(CMakeDependentOption)
 include(CMakePackageConfigHelpers)
 
+option(MORPHEUS_LINK_WITH_MOLD "Enable the mold linker" ON)
```

Comment:
Provide a CMake project-wide options to enable mode.  We default to enable so that when supported we will always enable it, when it is not supported we will fail gracefully back to the default linker.